core/audits/prioritize-lcp-image.js

```diff
@@ -47,42 +48,72 @@ class PrioritizeLcpImage extends Audit {
    *
    * @param {LH.Artifacts.NetworkRequest} request
    * @param {LH.Artifacts.NetworkRequest} mainResource
-   * @param {Array<LH.Gatherer.Simulation.GraphNode>} initiatorPath
+   * @param {InitiatorPath} initiatorPath
    * @return {boolean}
    */
   static shouldPreloadRequest(request, mainResource, initiatorPath) {
-    const mainResourceDepth = mainResource.redirects ? mainResource.redirects.length : 0;
-
     // If it's already preloaded, no need to recommend it.
     if (request.isLinkPreload) return false;
     // It's not a request loaded over the network, don't recommend it.
     if (NetworkRequest.isNonNetworkRequest(request)) return false;
     // It's already discoverable from the main document, don't recommend it.
-    if (initiatorPath.length <= mainResourceDepth) return false;
+    if (initiatorPath.length <= 2) return false;
```

Comment:
`initiatorPath` is now always at least `[lcpImage, mainResource]` and main resource redirects are never included, so this fixes the path length issue

Reply:
Can you add these details as a comment on this line (or somewhere nearby). perhaps in the jsdoc for `getLcpInitiatorPath`